install_config/registry/registry_known_issues.adoc

```diff
@@ -135,3 +135,21 @@ It means that your xref:../../install_config/registry/extended_registry_configur
 xref:../../install_config/registry/extended_registry_configuration.adoc#docker-registry-configuration-reference-storage[storage section], namely
 `*storage:delete:enabled*` set to *true*. Add them, re-deploy the registry, and
 repeat your image pruning operation.
+
+== Registry fails to deploy
+
+If the docker-registry pod fails to run and you see either the following error or no error at all.
+Verify that the corresponding service object exists.
+
+----
+$ oc get pods -n default -l docker-registry=default
+NAME                            READY     STATUS      RESTARTS   AGE
+docker-registry-2-l75vp         0/1       Error       4          1h
+$ oc logs docker-registry-2-l75vp
+time="2018-04-10T08:01:38Z" level=fatal msg="error parsing configuration file: configuration error in openshift.server.addr: REGISTRY_OPENSHIFT_SERVER_ADDR variable must be set when running outside of Kubernetes cluster"
```

Comment:
why is this the error when the docker-registry service does not exist? That env var doesn't have anything to do w/ the service.

Reply:
Yeah, this is a bit misleading. The `REGISTRY_OPENSHIFT_SERVER_ADDR` is the preferred env var for configuring this. If it's unset as well as some other deprecated vars with the same meaning, the [registry looks for the variables inherited from the associated service](https://github.com/miminar/image-registry/blob/master/pkg/dockerregistry/server/configuration/configuration.go#L355) (`"DOCKER_REGISTRY_SERVICE_HOST"`).

I'm putting the explanation in there.